nemo/collections/common/parts/preprocessing/collections.py

```diff
@@ -489,3 +481,253 @@ def _parse_item(self, line: str, manifest_file: str) -> Dict[str, Any]:
+                fed into the diarization system to merge the individual results.
+            *args: Args to pass to `SpeechLabel` constructor.
+            **kwargs: Kwargs to pass to `SpeechLabel` constructor.
+        """
+        self.round_digit = round_digit
+        self.emb_dict = emb_dict
+        self.clus_label_dict = clus_label_dict
+        self.seq_eval_mode = seq_eval_mode
+        self.pairwise_infer = pairwise_infer
+        audio_files, durations, rttm_files, offsets, target_spks_list, sess_spk_dicts, clus_spk_list, rttm_spk_list = (
+            [],
+            [],
+            [],
+            [],
+            [],
+            [],
+            [],
+            [],
+        )
+
+        for item in manifest.item_iter(manifests_files, parse_func=self.__parse_item_rttm):
+            if self.pairwise_infer:
+                if item['rttm_filepath']:
+                    base_scale_index = max(self.emb_dict.keys())
+                    _sess_spk_dict = self.emb_dict[base_scale_index][item['uniq_id']]['mapping']
+                    sess_spk_dict = {int(v.split('_')[-1]): k for k, v in _sess_spk_dict.items()}
+                    rttm_speaker_digits = [int(v.split('_')[1]) for k, v in _sess_spk_dict.items()]
+                else:
+                    sess_spk_dict = None
+                    rttm_speaker_digits = None
+
+                if self.seq_eval_mode:
+                    clus_speaker_digits = rttm_speaker_digits
+                else:
+                    clus_speaker_digits = sorted(list(set([x[2] for x in clus_label_dict[item['uniq_id']]])))
+
+                if len(clus_speaker_digits) == 1:
+                    spk_comb_list = [(0, 1)]
+                else:
+                    spk_comb_list = [x for x in combinations(clus_speaker_digits, 2)]
+
+            else:
+                target_spks = ((0, 1),)
+                spk_comb_list = [target_spks]
+                clus_speaker_digits = target_spks
+                rttm_speaker_digits = None
+                sess_spk_dict = None
+
+            for target_spks in spk_comb_list:
+                audio_files.append(item['audio_file'])
+                durations.append(item['duration'])
+                rttm_files.append(item['rttm_filepath'])
+                offsets.append(item['offset'])
+                target_spks_list.append(target_spks)
+                sess_spk_dicts.append(sess_spk_dict)
+                clus_spk_list.append(clus_speaker_digits)
+                rttm_spk_list.append(rttm_speaker_digits)
+
+        super().__init__(
+            audio_files,
+            durations,
+            rttm_files,
+            offsets,
+            target_spks_list,
+            sess_spk_dicts,
+            clus_spk_list,
+            rttm_spk_list,
+            *args,
+            **kwargs,
+        )
+
+    def get_speakers_from_rttm(self, rttm_path):
```

Comment:
I think this function is not used any where, we can remove this

Reply:
This function is removed. This was mistakenly included.

Reply:
removed